_tutorials/customize-geometries.md

```diff
@@ -0,0 +1,117 @@
+---
+layout: tutorial
+title: "Customizing Geometries"
+author: Alfonso Martínez (alfmarcua)
+tags: import lua
+---
+
+This tutorial will guide you through the process of customizing geometries in
+Nominatim. By customizing geometries, a more refined shape can be achieved for
+geographical elements, resulting in more precise and tailored search outcomes
+for specific use cases.
+
+This requirement is derived from a project under EUROSTAT aimed at developing
+[GISCO](https://ec.europa.eu/eurostat/web/gisco) map services to overcome
+political issues related to several geometries.
+
+## Prerequisites
+
+Before beginning, ensure that Nominatim is set up and operational on your
+system. Comprehensive installation instructions can be found on the [Nominatim
+website](https://nominatim.org/release-docs/develop/admin/Installation/). Also,
+make sure you have access to the PostgreSQL database that Nominatim is used for
+storing OpenStreetMap data.
+
+It is also needed a library for lua scripts to interact with the database. For
+that it may be installed with the following command:
+
+```shell
+apt install lua-sql-postgres
+```
+
+## Steps
+
+### 1. Import OSM Data
+
+With the lua script in place import OpenStreetMap (OSM) data into your Nominatim
+database. If not done already, consult the [Nominatim
+documentation](https://nominatim.org/release-docs/develop/admin/Import/) for
+precise instructions.
+
+### 2. Lua Script for Blocking Elements
+
+A custom Lua script in append mode will be developed to block elements with
+negative `osm_id` values. Since it will affect just in append mode it can be
+placed before or after the initial import of the database. For more information,
+consult the documentation for setting a [customize
+style](https://nominatim.org/release-docs/develop/customize/Import-Styles/).
+
+All negative `osm_id` values shall be retrieved from the database using
+`luasql.postgres` and then all negatives `osm_id` will be blocked only in append
+mode using `flex.process_tags`.
+
+```lua
+-- Import original process tags function from flex-base.lua
+local flex = require("flex-base")
+local original_process_tags = flex.process_tags
+
+-- Open a database connection to the database.
+local driver = require "luasql.postgres"
+local env = assert (driver.postgres())
+local con = assert (env:connect(""))
+
+-- Get all blocked items, which will not be processed 
+function get_blocked_ids()
+  -- This is only required for update
+  if osm2pgsql.mode == 'append' then
+    -- Get all blocked IDs from the database.
+    sql = [[ SELECT osm_id FROM placex
+             WHERE osm_id < 0
```

Comment:
You need to take the osm_type into account here because the ID space is per node/way/relation.

In this example you only change relations, so change the SQL `SELECT osm_id FROM placex WHERE osm_type = 'R' and osm_id < 0` and below....

Reply:
Done